samples/bluetooth/unicast_audio_server/src/main.c

```diff
@@ -205,15 +252,54 @@ static void stream_disconnected(struct bt_audio_stream *stream, uint8_t reason)
 	printk("Audio Stream %p disconnected (reason 0x%02x)\n", stream, reason);
 }
 
+
+#if defined(CONFIG_LIBLC3CODEC)
+
+static void stream_recv_lc3_codec(struct bt_audio_stream *stream, struct net_buf *buf)
+{
+	uint8_t bit_error_detect = 0;
+	uint8_t err;
+	/* TODO: If there is a way to know if the controller supports indicating errors in the
+	 *       payload one could feed that into bad-frame-indicator. The HCI layer allows to
+	 *       include this information, but currently there is no controller support.
+	 *       Here it is assumed that reveiving a zero-length payload means a lost frame -
+	 *       but actually it could just as well indicate a pause in the stream.
+	 */
+	uint8_t bad_frame_indicator = buf->len == 0 ? 1 : 0;
+
+	err = Lc3Decoder_run(lc3_decoder, buf->data, buf->len, bad_frame_indicator, audio_buf,
+			     sizeof(audio_buf)/2, &bit_error_detect);
```

Comment:
This callback is called in the RX thread. What is the runtime of this function? 

If it takes a while to decode the data, then perhaps it should be offloaded to a different thread.

Reply:
Good question. As we run on board native_posix I expect the HW to be rather powerful. My guess is a few 100 us. It might be worth having a task to do decoding, but it complicates the application quite a bit, as you would then also need a buffer to hold the data in case you receive multiple frames prior to completing the decode_run call.

Reply:
Might be worth adding a note at least then :)